packages/ckeditor5-watchdog/src/actionsrecorder.ts

```diff
@@ -0,0 +1,780 @@
+/**
+ * @license Copyright (c) 2003-2025, CKSource Holding sp. z o.o. All rights reserved.
+ * For licensing, see LICENSE.md or https://ckeditor.com/legal/ckeditor-licensing-options
+ */
+
+/**
+ * @module watchdog/actionsrecorder
+ */
+
+/* istanbul ignore file -- @preserve */
+
+// eslint-disable-next-line ckeditor5-rules/no-cross-package-imports
+import {
+	Plugin,
+	type Command,
+	type Editor
+} from '@ckeditor/ckeditor5-core';
+
+import type {
+	ViewTypeCheckable,
+	ModelTypeCheckable
+} from '@ckeditor/ckeditor5-engine';
+
+import type {
+	ActionsRecorderEntry,
+	ActionsRecorderEntryEditorSnapshot,
+	ActionsRecorderFilterCallback,
+	ActionsRecorderErrorCallback,
+	ActionsRecorderMaxEntriesCallback
+} from './actionsrecorderconfig.js';
+
+import { isPlainObject } from 'es-toolkit/compat';
+
+/**
+ * A plugin that records user actions and editor state changes for debugging purposes. It tracks commands execution, model operations,
+ * UI interactions, and document events. It just collects data locally, and does not send it anywhere, integrator is responsible
+ * for gathering data from this plugin for further processing.
+ *
+ * By default, plugin stores latest 1000 action entries. Integrator can register an `onError` callback to collect those entries
+ * along with the caught exception, augment those data with application data such as page-id or session-id, depending on the application,
```

Comment:
The collected entries contain exceptions. IMO, "along with the caught exception" is confusing (especially in singular form, when talking about many entries - they have individual `error` fields). The rest of the description (starting with "augment") should be a separate statement(s) and we need to make it clear that we're done with describing the callback and we're back to integrator duties.

Reply:
Addressed in d757c29.